src/Authentication.Abstractions/Interfaces/IAuthenticationFactory.cs

```diff
@@ -133,6 +133,15 @@ IAccessToken Authenticate(
         /// </summary>
         /// <param name="account">The account to remove credentials for</param>
         /// <param name="tokenCache">The TokenCache to remove credentials from</param>
+        [Obsolete("RemoveUser is deprecated, please use RemoveUser with Azure environment instead.", true)]
         void RemoveUser(IAzureAccount account, IAzureTokenCache tokenCache);
+
+
+        /// <summary>
+        /// Remove any stored credentials for the given user and the Azure environment used.
+        /// </summary>
+        /// <param name="account">The account to remove credentials for</param>
+        /// <param name="authority">The Microsoft Entra authority</param>
+        void RemoveUser(IAzureAccount account, string authority);
```

Comment:
Would it make sense if we require the IAzureEnvironment interface rather than just the authority?

Reply:
Done